athena/decode_main.py

```diff
@@ -29,41 +30,23 @@
 )
 
 
-def decode(jsonfile, n=1, log_file=None):
+def decode(jsonfile):
     """ entry point for model decoding, do some preparation work """
     p, model, _, checkpointer = build_model_from_jsonfile(jsonfile)
+    if 'decode_log' in p.decode_config:
+        decode_log = p.decode_config['decode_log']
+        log_dir = os.path.dirname(decode_log)
+        log_name = os.path.basename(decode_log)
+        logging.get_absl_handler().use_absl_log_file(log_name, log_dir)
+    mode = 'avg'
+    if 'load_mode' in p.decode_config:
+        mode = 'best' if 'ckpt_mode' == 'best' else 'avg'
+    checkpointer.restore_from_best(mode=mode)
```

Comment:
Pass 'avg' to restore_from_best is weird. We should add another function like 'restore_from_average' in checkpointer.